Bugfixing

wintuxx opened the pull request:
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the [**CONTRIBUTING**](Contributing.md) document.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?


### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully ran tests with your changes locally?

added all changes from @jakermx and @CastagnaIT

Trying to keep this project alive

wintuxx committed:
Add files via upload

wintuxx committed:
Update converter.py

https://github.com/asciidisco/plugin.video.netflix/pull/519#issuecomment-447636788

wintuxx committed:
Update README.md

CastagnaIT committed:
Fixed default audio language track selection

wintuxx committed:
Merge pull request #1 from CastagnaIT/release/0.14.0

Fixed default audio language track selection

CastagnaIT committed:
Updated translation for it-it

wintuxx committed:
Merge pull request #2 from CastagnaIT/release/0.14.0

Updated translation for it-it

CastagnaIT committed:
fix mistake translation for it-it

wintuxx committed:
Merge pull request #3 from CastagnaIT/release/0.14.0

fix mistake translation for it-it (sorry)

CastagnaIT committed:
Fixed setting menu Views selectable

*note not all View settings works (ex. Movies), need more fixes behind the code

wintuxx committed:
Merge pull request #4 from CastagnaIT/release/0.14.0

Fixed setting menu Views selectable

wintuxx commented:
okay there are many issues to fix. i revolk the pull

CastagnaIT commented:
My changes do not break functionality it can be added those commits.
The customizations at the views were not working even before.
To fix view settings definitively, i'm working to centralize the configuration of the main menu management to facilitate future changes (views, context, names, etc.) at least in part.
Not knowing the project well, it's not easy to understand everything...

Besides that, are there other serious problems?

wintuxx commented:
yes i can confirm your changes are working. 
I'm not that familiar with python yet. I let myself be confused by the test.
I open the pull again and then we'll see

CastagnaIT committed:
Fixed wrong size of control container of skip modal dialog

*causes off-screen positioning when there is long translated text

CastagnaIT committed:
Fixed mastertoken expired error (not android)

Cause of error: 'Only byte strings can be passed to C code' every time
you start a video

*now msl data renewal works, maybe you can implement it better than that
*on android the system is different, it does not seem to me to be
affected

wintuxx committed:
Merge pull request #5 from CastagnaIT/release/0.14.0

Merge

CastagnaIT commented:
Please wait for the last commit is not definitive

wintuxx committed:
Revert "Merge"

CastagnaIT committed:
Fixed msldata renewal on expired mastertoken

Cause of errors:
Widevine CryptoSession getKeyRequest failed! (on android)
Only byte strings can be passed to C code (on windows)

CastagnaIT committed:
Fix mslhandler error on new installations

it is necessary to take into account that in new installations there is
no esn, so perform a key handshake isn't possible, so nothing msl_data
saved.

wintuxx committed:
Merge pull request #6 from wintuxx/revert-5-release/0.14.0

Revert "Merge"

wintuxx committed:
Merge pull request #7 from CastagnaIT/release/0.14.0

Merge

CastagnaIT committed:
v2 license request + some fixes from MSLv2 impl.

refs commit:
v2 license request / Add h264 profileLevel40 streams
c2d113597bfee4ca1a4c4e4ec4cfca3ed158ad5d

MSLv2 implementation
c2307770dadc99da6f463d08b2e5bcfa6d162eff

CastagnaIT committed:
Add some h264 profileLevel40 streams

CastagnaIT committed:
Fixed default audio language track selection

CastagnaIT committed:
Updated translation for it-it

CastagnaIT committed:
Fixed setting menu Views selectable

*note not all View settings works (ex. Movies), need more fixes behind the code

CastagnaIT committed:
Fixed wrong size of control container of skip modal dialog

*causes off-screen positioning when there is long translated text

CastagnaIT committed:
Fixed msldata renewal on expired mastertoken

Cause of errors:
Widevine CryptoSession getKeyRequest failed! (on android)
Only byte strings can be passed to C code (on windows)

CastagnaIT committed:
Fix mslhandler error on new installations

it is necessary to take into account that in new installations there is
no esn, so perform a key handshake isn't possible, so nothing msl_data
saved.

wintuxx committed:
Merge branch 'release/0.14.0' of git://github.com/CastagnaIT/plugin.video.netflix into CastagnaIT-release/0.14.0

wintuxx committed:
git push origin release/0.14.0Merge branch 'CastagnaIT-release/0.14.0' into release/0.14.0

corrected Merge from CastagnaIT-release/0.14.0

wintuxx committed:
manual copy from CastagnaIT/plugin.video.netflix

Fixing issues to my fork because merging

wintuxx committed:
Fixing Issues

wintuxx committed:
manual copy from CastagnaIT/plugin.video.netflix

wintuxx committed:
manual copy from CastagnaIT/plugin.video.netflix

wintuxx committed:
Fixing command merge

ccope commented on resources/lib/services/msl/converter.py:
```diff
@@ -20,11 +20,16 @@ def convert_to_dash(manifest):
     root = _mpd_manifest_root(duration)
     period = ET.SubElement(root, 'Period', start='PT0S', duration=duration)
     protection = _protection_info(manifest)
+<<<<<<< HEAD
```
This file has lots of unresolved merge conflicts

wintuxx replied:
fixed

ccope commented on resources/lib/services/msl/msl_handler.py:
```diff
@@ -26,9 +26,14 @@
 
 CHROME_BASE_URL = 'https://www.netflix.com/nq/msl_v1/cadmium/'
 ENDPOINTS = {
+<<<<<<< HEAD
```
This file also has unresolved merge conflicts

wintuxx replied:
fixed

ccope commented on resources/lib/services/msl/request_builder.py:
```diff
@@ -72,16 +73,21 @@ def _headerdata(self, esn, is_handshake=False, is_key_request=False,
             'handshake': is_handshake,
             'nonreplayable': False,
             'capabilities': {
+<<<<<<< HEAD
```
This file as well.

wintuxx replied:
thx for your sugesstions.
I have fixed them all allready

wintuxx committed:
free space after #

CastagnaIT committed:
Revert back some changes v2 license seem not fully support gzip

in decrypt_chunks() fixed return data, the decrypted_payload content is different
between license and manifest

CastagnaIT committed:
decrypt_chunks: fixed UnicodeDecodeError

error: UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 11037: ordinal not in range(128)
it happens with languages containing special characters such as fr_FR

CastagnaIT committed:
msl_handler: temporany removed 'success' check

*for now removed, i have not been able to fix it, i can not get a wrong
manifest or license, needs more tests

wintuxx committed:
Merge pull request #11 from CastagnaIT/release/0.14.0

Update

CastagnaIT committed:
Version bump (beta19)

wintuxx committed:
Merge pull request #12 from CastagnaIT/release/0.14.0

Version bump (beta19)

CastagnaIT commented:
I understand the intention to keep caphm up to date, but this is counterproductive, if caphm decides to continue the development he will take the commits directly from my repository not from this mess...
